### PR TITLE
Add example of Custom Kafka Headers

### PR DESCRIPTION
**Pull request by oleg-rakitin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,4 +78,50 @@ Not supported
...
+| storage-net-node-port | integer | 9092 | The common port number to access the storage nodes, may be overriden adding the port number to the storage-driver-addrs, for example: "127.0.0.1:9020,127.0.0.1:9022,..." |
+
+## Custom Kafka Headers
+
+Scenario example:
+
+```javascript
+var customKafkaHeadersConfig = {
+    "storage" : {
+        "net" : {
```
Comment (akurilov) [inline README.md]: let's use `storage-driver-request-headers` either `storage-driver-create-headers` option
also, the doc is updated, but the option is not added into the defaults/schema yaml files
moreover, the declared configuration option should be handled by the driver's code

**Review by akurilov — DISMISSED** [review]

**Review by AndreyKoltsov1997 — DISMISSED** [review]
What about the record key?

File: src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java [hunk 2]
```diff
@@ -31,6 +37,22 @@ public KafkaStorageDriver(
...
+      final var headerValue = header.getValue();
+      if (headerKey.contains(ASYNC_MARKER)
+              || headerKey.contains(SYNC_MARKER)
+              || headerKey.contains(INIT_MARKER)
+              || headerValue.contains(ASYNC_MARKER)
+              || headerValue.contains(SYNC_MARKER)
+              || headerValue.contains(INIT_MARKER)) {
+        dynamicHeaders.put(headerKey, headerValue);
+      } else {
+        throw new IllegalArgumentException("Unsupported key");
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java]: the constant headers should be used also

**Review by akurilov — DISMISSED** [review]

File: src/main/resources/config/defaults-storage-driver-kafka.yml [hunk 3]
```diff
@@ -7,6 +7,10 @@ storage:
     create:
       key:
         enabled: false
+      headers:
+        Connection: Keep-Alive
```
Comment (akurilov) [inline src/main/resources/config/defaults-storage-driver-kafka.yml]: this is HTTP-specific headers which are not suitable for kafka
let's leave the headers empty

**Review by akurilov — CHANGES_REQUESTED** [review]

File: src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java [hunk 4]
```diff
@@ -65,6 +72,21 @@ public KafkaStorageDriver(
     throws IllegalConfigurationException {
     super(testStepId, dataInput, storageConfig, verifyFlag, batchSize);
     var driverConfig = storageConfig.configVal("driver");
+	final var headersMap = driverConfig.<String>mapVal("headers");
+    for (final var header : headersMap.entrySet()) {
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java]: the default value will cause NullPointerException here

File: src/main/resources/config/defaults-storage-driver-kafka.yml [hunk 5]
```diff
@@ -7,6 +7,7 @@ storage:
     create:
       key:
         enabled: false
+      headers: null
```
Comment (akurilov) [inline src/main/resources/config/defaults-storage-driver-kafka.yml]: https://stackoverflow.com/questions/33510094/syntax-for-empty-dictionary-in-yaml

**Review by akurilov — CHANGES_REQUESTED** [review]

**Review by akurilov — APPROVED** [review]

**Pull request merged**